src/winkerberos.c

```diff
@@ -437,8 +437,160 @@ sspi_client_clean(PyObject* self, PyObject* args) {
+#else
+void destruct_channel_bindings_struct(void* o) {
+    SecPkgContext_Bindings* context_bindings = (SecPkgContext_Bindings *)o;
+#endif
+    if (context_bindings != NULL) {
+        free(context_bindings->Bindings);
+        free(context_bindings);
+    }
+}
+
+PyDoc_STRVAR(sspi_channel_bindings_doc,
+"channelBindings(**kwargs)\n"
+"\n"
+"Builds a SecPkgContext_Bindings struct and returns a pointer to this struct.\n"
+"This struct can then be passed into `authGSSClientStep` under the\n"
+"channel_bindings kwarg argument. The SecPkgContext_bindings object\n"
+"destroys itself when it is reclaimed. While the parameters supported by this\n"
+"method match the structure of the GSSAPI gss_channel_bindings_t\n"
+"https://docs.oracle.com/cd/E19455-01/806-3814/overview-52/index.html\n"
+"Windows only uses the application_data parameter for now. The other\n"
+"parameters are kept for compatibility with ccs-pykerberos and future\n"
+"compatibility.\n"
+"\n"
+"If using an endpoint over TLS like IIS with Extended Protection or WinRM\n"
+"with the CbtHardeningLevel set to Strict then you can use this method to\n"
+"build the channel bindings structured required for a successful\n"
+"authentication. When calling this method in this scenario you can follow\n"
+"RFC5929 - Channel Bindings for TLS where you pass in the value\n"
+"tls-server-end-point:{cert-hash} into the application_data argument.\n"
+"There are numerous ways in which you can access the certificate hash but\n"
+"if you are using requests you can get it from a response object like so\n"
+"    if sys.version_info > (3, 0):\n"
+"        socket = response.raw._fp.fp.raw._sock\n"
+"    else:\n"
+"        socket = response.raw._fp.fp._sock\n"
+"    server_certificate = socket.getpeercert(True)\n"
+"    certificate_hash = hashlib.sha256(server_certificate)\n"
+"    certificate_digest = base64.b16decode(certificate_hash)\n"
+"    application_data = b'tls-server-endpoint:%s' % certificate_digest\n"
+"    channel_bindings = winkerberos.channelBindings(application_data=application_data)\n"
+"\n"
+"There are numerous rules around how to derive the hash from the certificate\n"
+"but in most cases it would be the SHA256 hash."
+"\n"
+":Parameters:\n"
+"  - `initiator_addrtype`: Optional int of the initiator address type,\n"
+"    default is GSS_C_AF_UNSPEC (0).\n"
+"  - `initiator_address`: Optional byte string of the initiator address.\n"
+"  - `acceptor_addrtype`: Optional int of the acceptor address type,\n"
+"    default is GSS_C_AF_UNSPEC (0).\n"
+"  - `acceptor_address`: Optional byte string of the acceptor address.\n"
+"  - `application_data`: Optional byte string that should be set under the\n"
+"    application_data field. An example of this would be\n"
+"    'tls-server-endpoint:{cert-hash}' where {cert-hash} is the hash of the\n"
+"    server's certificate.\n"
+"\n"
+":Returns: SecPkgContext_Bindings* where SecPkgContext_Bindings is the\n"
```

Comment:
I would just say something like...

"Returns: an opaque value to be passed to the channel_bindings parameter of authGSSClientStep"

That language is similar to how we describe the context object returned by authGSSClientInit.